src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -81,6 +84,25 @@ internal sealed class AggregatorStore
             this.tagKeysInteresting = hs;
             this.tagsKeysInterestingCount = hs.Count;
         }
+
+        // If the switch is not set at all or if it's explicitly set to false
+        if (AppContext.TryGetSwitch("OTel.Dotnet.EmitMetricOverflowAttribute", out bool shouldEmitOverflowAttribute) == false ||
+            shouldEmitOverflowAttribute == false)
+        {
+            emitOverflowAttribute = false;
+        }
+        else if (this.maxMetricPoints > 1)
+        {
+            // We need at least two metric points. One is reserved for zero tags and the other one for overflow attribute
+
+            emitOverflowAttribute = true;
+
+            // Setting this to as we would reserve the metricPoints[1] for overflow attribute.
+            // Newer attributes should be added starting at the index: 2
+            this.metricPointIndex = 1;
+
+            this.metricPoints[1] = new MetricPoint(this, this.aggType, new KeyValuePair<string, object>[] { new("otel.metric.overflow", true) }, this.histogramBounds, this.exponentialHistogramMaxSize, this.exponentialHistogramMaxScale);
```

Comment:
Should we do this when actually needed (i.e the very first time?), similar to how we initialize zero tag
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Metrics/AggregatorStore.cs#L185

Reply:
Great suggestion! Thanks for that. I have updated the code.